src/fix_downleveled_decorators.ts

```diff
@@ -0,0 +1,60 @@
+import * as ts from "typescript";
+
+/**
+ * This is an attempt to fix the downleveled decorators so Closure doesn't have
+ * trouble with them. The problem is that when `experimentalDecorators` is
+ * enabled, we TSC ends up converting a class decorator like this:
+ *
+ * @classDecorator
+ * export class Person { ... }
+ *
+ * to this:
+ *
+ * let Person = class Person { ... };
+ *
+ * as well as some calls to __decorate(Person, ...)
+ *
+ * The problem is that this causes Closure Compiler to fail with this error:
+ * ERROR - [JSC_CANNOT_CONVERT_YET] Transpilation of 'Classes with possible name shadowing' is not yet implemented.
+ * 21| let Person = class Person {
+ *                  ^^^^^^^^^^^^^^
+ *
+ * As previously mentioned, this transformer is an _attempt_, it doesn't actually work it seems.
+ * I'm not sure if tsickle can even affect this at this point in the process or not.
```

Comment:
It seems like the `transformDecoratorsOutputForClosurePropertyRenaming` is able to update the ast after decorators are converted to calls to `tslib.__decorate`, so I would expect we should be able to modify things at the same point

Reply:
It turns out if you don't `return` something... it does nothing 🤦🏼